### PR TITLE
Periodic system timer & esp32s2 fixes & edge interrupt fixes

### LINKED ISSUE #140: [esp32s2] systimer example doesn't work

### PR DESCRIPTION
**Pull request by MabezDev**
Closes #140 
Partially closes (the systimer part) #123

### CODE REVIEW

**Comment by MabezDev** [comment]
FYI when you run the esp32s2 example you'll notice the TARGET1 & TARGET2 interrupts fire at the same time for each of the respective alarms they are associated with at that level. This is not a bug, but just the nature of the interrupts being edge interrupts instead of level (thankfully they're level based in the c3 & s3).

File: esp32s2-hal/examples/systimer.rs [hunk 1]
```diff
@@ -41,8 +41,8 @@ fn main() -> ! {
 
     let syst = SystemTimer::new(peripherals.SYSTIMER);
```
Comment (jessebraham) [inline esp32s2-hal/examples/systimer.rs:43]: Could you add the print statement from the other examples here please?

```suggestion
esp_println::println!("SYSTIMER Current value = {}", SystemTimer::now());
```

**Review by jessebraham — CHANGES_REQUESTED** [review]
Just one little change I'd like please!

**Comment by bjoernQ** [comment]
Awesome! Didn't test for S2 yet - waiting for the toolchain builds
One minor thing; Maybe we should lower the targets for alarm1 and 2 in the examples since it takes ages to hit them

**Comment by bjoernQ** [comment]
Also works fine on ESP32-S2 for me 🎉

**Review by bjoernQ — APPROVED** [review]
LGTM - just as mentioned maybe adjusting the examples might be a good idea

**Review by bjoernQ — APPROVED** [review]
LGTM

**Review by jessebraham — APPROVED** [review]
Thanks for making that change, LGTM!

**Pull request merged**